src/backend/joanie/tests/core/test_api_order.py

```diff
@@ -648,6 +648,53 @@ def test_api_order_read_list_filtered_by_state_validated(self, _mock_thumbnail):
             },
         )
 
+    @mock.patch.object(
+        fields.ThumbnailDetailField,
+        "to_representation",
+        return_value="_this_field_is_mocked",
+    )
+    def test_api_order_read_list_filtered_by_multiple_states(self, _mock_thumbnail):
+        """Authenticated user should be able to retrieve its validated orders."""
+        [product_1, product_2, product_3, product_4] = factories.ProductFactory.create_batch(4, price=0.00)
+        user = factories.UserFactory()
+
+        # User purchases the product 1 as its price is equal to 0.00€,
+        # the order is directly validated
+        factories.OrderFactory(
+            owner=user, product=product_1, state=enums.ORDER_STATE_VALIDATED
+        )
+        factories.OrderFactory(
+            owner=user, product=product_2, state=enums.ORDER_STATE_PENDING
+        )
+        factories.OrderFactory(
+            owner=user, product=product_3, state=enums.ORDER_STATE_SUBMITTED
+        )
+
+        # User purchases the product 2 then cancels it
+        factories.OrderFactory(
+            owner=user, product=product_4, state=enums.ORDER_STATE_CANCELED
+        )
```

Comment:
you should be able to remove the line on top creating the 4 products and just write:
```suggestion
        factories.OrderFactory(
            owner=user, state=enums.ORDER_STATE_VALIDATED
        )
        factories.OrderFactory(
            owner=user, state=enums.ORDER_STATE_PENDING
        )
        factories.OrderFactory(
            owner=user, state=enums.ORDER_STATE_SUBMITTED
        )

        # User purchases the product 2 then cancels it
        factories.OrderFactory(
            owner=user, state=enums.ORDER_STATE_CANCELED
        )
```